src/AutoMapper/Mappers/DynamicMappers.cs

```diff
@@ -62,9 +62,9 @@ protected void SetDynamically(MemberInfo member, object target, object value)
 
     public class FromDynamicMapper : DynamicMapper
     {
-        public override bool IsMatch(ResolutionContext context)
+        public override bool IsMatch(TypePair context)
         {
-            return context.SourceValue.IsDynamic() && !context.DestinationType.IsDynamic();
+            return context.SourceType.IsDynamic() && !context.DestinationType.IsDynamic();
```

Comment:
I used SourceValue so I wouldn't have to check for null. We should check what happens when you map null.